lib/libraries/c3/extensions/paginating-legend.js

```diff
@@ -0,0 +1,234 @@
+var d3 = require('d3');
+
+var isDateString = require('../../../utils/assert-date-string');
+
+module.exports = function(cols){
+  var self = this,
+      chart = this.view._artifacts.c3,
+      columns = [],
+      domNode = this.el().querySelector('.' + this.theme() + '-rendering'),
+      legendWidth = 120;
+
+  var pagination = this.view._artifacts.pagination = {
+    hidden: [],
+    labels: [],
+    position: 0,
+    range: Math.round((domNode.offsetHeight - 78) / 20),
+    total: 0
+  };
+
+  // Reduce threshold for donut/pie charts
+  if (this.type() === 'donut' || this.type() === 'pie') {
+    pagination.range = pagination.range >= 5 ? 5 : pagination.range;
+  }
+
+  for (var i = 0; i < cols.length; i++) {
+    if (cols[i][0] !== 'x' && !isDateString(cols[i][1])) {
+      columns.push(cols[i][0]);
+    }
+  }
+
+  var legendEl = d3.select(domNode)
+    .append('svg')
+    .attr({
+      'class': 'keen-c3-legend',
+      'height': domNode.offsetHeight - 10,
+      'width': legendWidth
+    })
+    .style({
+      'right': -legendWidth + 'px',
+      'top': '10px'
+    });
+
+  var legendItems = legendEl
+    .append('g')
+    .attr('class', 'keen-c3-legend-items');
+
+  var paginateElOffset = 20 * pagination.range;
+  var paginateEl = legendEl
+    .append('g')
+    .attr({
+      'class': 'keen-c3-legend-pagination',
+      'transform': function(){
+        return 'translate(2, ' + paginateElOffset + ')'
+      }
+    });
+
+  paginateData();
+
+  function paginateData(){
```

Comment:
👍  Let's chat more about it, but I don't think we need to hold up this PR for it.